content/docs/instrumenting/pushing.md

```diff
@@ -26,6 +26,56 @@ class.
 
  * For use from Ruby see the [Pushgateway documentation](https://github.com/prometheus/client_ruby#pushgateway).
 
+## Should I be using the Pushgateway?
+
+**We only recommend using the Pushgateway in certain limited cases.** There are
+several pitfalls when blindly using the Pushgateway instead of Prometheus's
+usual pull model for general metrics collection:
+
+* When monitoring multiple instances through a single Pushgateway, the
+  Pushgateway becomes both a single point of failure and a potential
+  bottleneck.
+* You lose Prometheus's automatic instance health monitoring via the `up`
+  metric (generated on every scrape).
+* The Pushgateway never forgets series pushed to it and will expose them to
+  Prometheus forever unless those series are manually deleted via the
+  Pushgateway's API.
+
+The latter point is especially relevant when multiple instances of a job
+differentiate their metrics in the Pushgateway via an `instance` label or
+similar. Metrics for an instance will then remain in the Pushgateway even if
+the originating instance is renamed or removed. This is because the lifecycle
+of the Pushgateway as a metrics cache is fundamentally separate from the
+lifecycle of the processes that push metrics to it. Contrast this to
+Prometheus's usual pull-style monitoring: when an instance disappears
+(intentional or not), its metrics will automatically disappear along with it.
+When using the Pushgateway, this is not the case, and you would now have to
+delete any stale metrics manually or automate this lifecycle synchronization
+yourself.
+
+**Usually, the only valid use case for the Pushgateway is for monitoring
+service-level batch jobs which do not run long enough to be scraped reliably.**
+A "service-level" batch job is one which is not semantically related to a
+specific machine or job instance (for example, a batch job that deletes a
+number of users for an entire service). Such a job's metrics should not include
+a machine or instance label to decouple the lifecycle of specific machines or
+instances from the pushed metrics. This decreases the burden for managing stale
+metrics in the Pushgateway.
+
+For batch jobs that are related to a machine (such as automatic
+security update cronjobs or configuration management client runs), expose the
+resulting metrics using the [Node Exporter's](https://github.com/prometheus/node_exporter)
+textfile module instead of the Pushgateway.
+
+For batch jobs that neither belong to a specific machine nor to an entire
+service (i.e. they should expose metrics about individual instances of a job),
+consider converting them into long-running processes that can be scraped.
```

Comment:
If there's an instance the batch job is associated with, then rather than making the batch job a new daemon beside the existing daemon it should either be folded in or take care of exposing the metrics.